glean/metrics.yaml

```diff
@@ -1637,3 +1637,179 @@ sample:
       language_code:
         description: The language code chosen.
         type: string
+
+  speed_test_completed:
+    type: event
+    lifetime: ping
+    send_in_pings:
+      - main
+    description: |
+      Speed test was completed.
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/VPN-3630
+    data_reviews:
+      - https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5702#issuecomment-1397761161
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - mcleinman@mozilla.com
+    expires: never
+    extra_keys:
+      speed:
+        description: |
+          The speed bucket that the test resulted in: Slow | Medium | Fast
+        type: string
+
+  bottom_navigation_bar_click:
+    type: event
+    lifetime: ping
+    send_in_pings:
+      - main
+    description: |
+      The user has tapped a bottom navigation bar button.
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/VPN-3630
+    data_reviews:
+      - https://github.com/mozilla-mobile/mozilla-vpn-client/pull/5702#issuecomment-1397761161
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - mcleinman@mozilla.com
+    expires: never
+    extra_keys:
+      bar_button:
+        description: |
+          The name of the bar button tapped:
+          ScreenMessaging | ScreenHome | ScreenSettings | Unknown
+        type: string
+
+  authentication_error:
```

Comment:
How is this metric different from the myriad of authentication error metrics we already have? https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn?page=1&search=authentication

Reply:
From conversations with Santiago, he wanted to be able to drill into where the failure happened. The extra keys for this metric (Timeout | IncorrectPassword | InvalidEmail | InvalidOrExpiredVerificiationCode | InvalidTOTP | InvalidUnblockCode) are what makes this different. 

Do you think we should consider removing the old one? (If it's currently used for analysis, this would obviously be subpar.) Or adding these keys to it? Or something else?

Reply:
I'm a bit confused. What do you mean by "where the failure happened"? The extra keys are more about _which_ failure happened, no? If so, that is already answered by https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn/metrics/sample_authentication_inapp_error if I understand correctly. See the "error" extra key. If anything, `authentication_inapp_error` even provides more info about the error.

Reply:
Santiago was most concerned with "what error does the user see", and seemingly less concerned about what is happening technically behind the scenes. (A metric that gives both "what users see" and "what is happening technically" would be best.) I'll acknowledge that I didn't see this existing metric until you pointed it out in your earlier comment, but also - looking at the flow, it doesn't seem like it captures many things that Santiago is interested in - see case 183 and 103 for example. A whole bunch of that switch statement doesn't end up logging something in glean.

I can add add glean logging for all the additional errors not currently covered, then provide a translation somewhere of which error numbers map to which user-shown errors. Over time as code changes are made, that's likely to diverge and not be kept up to date. 

It seems like we have two approaches here:
1) (How it's currently written.) Log technical errors and user-facing errors separately in glean, with different metrics. It's a little duplicative, but it stays up to date with no engineering work, and it's easier to analyze.
2) Log both in one glean metric. Do some post-processing to separate all the technical details into the user-facing errors. (AKA: Data analysts have additional code, rather than our codebase as in the #1.) Try to provide comments in code so that we don't forget to update that technical-error-to-user-error-translation guide when code changes in future years, but it's higher risk that the user-facing error metric will fall out of date at some point.

I think I have a slight inclination towards #1, as it's easier for analysis and more future-proof. (Part of this project is about making it easier for PMs and others to run ad-hoc queries.) I definitely don't feel strongly about this though - despite the length of what I just wrote (sorry) - and so if our approach is usually closer to number 2, let me know and I'll implement it.

Reply:
> Santiago was most concerned with "what error does the user see", and seemingly less concerned about what is happening technically behind the scenes.

I would argue that you can infer what the user is seeing from what is happening behind the scenes. But I won't block this anymore.